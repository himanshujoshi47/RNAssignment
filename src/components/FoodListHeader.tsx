import React, {memo, useState} from 'react';
import {View, StyleSheet, Text, TextInput} from 'react-native';
import {Icon} from 'react-native-elements';
import R from '../res';

interface FoodListHeaderProps {
  didCloseFoodListModal: () => void;
  performSearch: (text: string) => void;
}

const FoodListHeader: React.FC<FoodListHeaderProps> = ({
  didCloseFoodListModal,
  performSearch,
}) => {
  const [searchText, setSearchText] = useState<string>('');

  function onSearch(text: string) {
    let _originalSearchText = text;
    let trimmedText = text.trim();
    if (trimmedText.length === 0) {
      _originalSearchText = trimmedText;
    }
    performSearch(_originalSearchText);
    setSearchText(_originalSearchText);
  }

  return (
    <View>
      <Icon
        name="close"
        color={R.colors.stroke.black}
        size={R.dimensions.dimen_44}
        testID={R.id.CLOSE_FOOD_LIST_MODAL}
        style={localStyles.closeIcon}
        onPress={() => {
          didCloseFoodListModal();
        }}
      />
      <Text testID={R.id.APPROVED_FOOD_LIST_TITLE} style={localStyles.title}>
        {R.strings.APPROVED_FOOD_LIST}
      </Text>
      <View style={localStyles.searchContainer}>
        <Icon
          name={'search'}
          color={R.colors.text.secondary}
          size={R.dimensions.dimen_24}
        />
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          value={searchText}
          onChangeText={queryText => onSearch(queryText)}
          placeholder={R.strings.SEARCH_PLACEHOLDER}
          placeholderTextColor={R.colors.text.secondary}
          testID={R.id.FOOD_LIST_SEARCH_INPUT}
          style={localStyles.searchInput}
        />
      </View>
    </View>
  );
};

const localStyles = StyleSheet.create({
  closeIcon: {
    justifyContent: 'flex-start',
    width: R.dimensions.dimen_44,
    height: R.dimensions.dimen_44,
  },
  title: {
    fontSize: R.dimensions.dimen_24,
    marginVertical: R.dimensions.dimen_24,
    fontWeight: '500',
    color: R.colors.text.primary,
  },
  searchContainer: {
    backgroundColor: R.colors.background.whiteSoft,
    padding: R.dimensions.dimen_16,
    marginBottom: R.dimensions.dimen_16,
    borderRadius: R.dimensions.dimen_4,
    flexDirection: 'row',
  },
  searchInput: {
    flex: 1,
    color: R.colors.text.primary,
    marginLeft: R.dimensions.dimen_8,
    fontSize: R.dimensions.dimen_18,
    fontWeight: '500',
  },
});

export default memo(FoodListHeader);
