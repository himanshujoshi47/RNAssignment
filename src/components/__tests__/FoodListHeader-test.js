import React from 'react';
import renderer, {act} from 'react-test-renderer';
import FoodListHeader from '../FoodListHeader';
import R from '../../res';

jest.useFakeTimers();

describe('renders and behave correctly', () => {
  let props = {
    didCloseFoodListModal: jest.fn(),
    performSearch: jest.fn(),
  };

  const foodListHeader = renderer.create(<FoodListHeader {...props} />);
  const foodListHeaderInst = foodListHeader.root;

  it('renders correctly', () => {
    expect(foodListHeader).toMatchSnapshot();
  });

  it('should have required elements', () => {
    expect(
      foodListHeaderInst.findByProps({testID: R.id.CLOSE_FOOD_LIST_MODAL}),
    ).toBeTruthy();
    expect(
      foodListHeaderInst.findByProps({testID: R.id.APPROVED_FOOD_LIST_TITLE}),
    ).toBeTruthy();
    expect(
      foodListHeaderInst.findByProps({testID: R.id.FOOD_LIST_SEARCH_INPUT}),
    ).toBeTruthy();
  });

  it('expand/collapse category on select', () => {
    act(() => {
      foodListHeaderInst
        .findByProps({testID: R.id.CLOSE_FOOD_LIST_MODAL})
        .props.onPress();
    });

    expect(props.didCloseFoodListModal).toBeCalled();
  });

  it('should search while typing text in search bar', () => {
    act(() => {
      foodListHeaderInst
        .findByProps({testID: R.id.FOOD_LIST_SEARCH_INPUT})
        .props.onChangeText('loin');
    });

    expect(props.performSearch).toBeCalledWith('loin');
  });
});
