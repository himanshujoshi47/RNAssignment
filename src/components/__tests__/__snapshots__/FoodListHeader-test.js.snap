// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and behave correctly renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#222222",
        "fontSize": 24,
        "fontWeight": "500",
        "marginVertical": 24,
      }
    }
    testID="approved-food-list-title"
  >
    Approved Foods List
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF4D",
        "borderRadius": 4,
        "flexDirection": "row",
        "marginBottom": 16,
        "padding": 16,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Try searching fat, sauces names..."
      placeholderTextColor="#CCCCCC"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#222222",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "500",
          "marginLeft": 8,
        }
      }
      testID="food-list-search-input"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
